### PR TITLE
Re-implement shadow's "component" method to correctly

### LINKED ISSUE #4119: Unable to build against DHC's psk-authentication-provider
**Description**

I'm building DnD locally in Vermilion against DHC 0.26.0, which i've published locally (since it's not released yet). I'm able to build fine when running jar, but on distTar, i'm getting errors like:
Execution failed for task ':startScripts'.
> Error while evaluating property 'relativeClasspath' of task ':startScripts'
   > Could not resolve all files for configuration ':runtimeClasspath'.
      > Could not find io.deephaven:authentication:0.26.0.
        Searched in the following locations:
          - file:/Users/afarhangi/.m2/repository/io/deephaven/authentication/0.26.0/authentication-0.26.0.pom
          - https://repo.maven.apache.org/maven2/io/deephaven/authentication/0.26.0/authentication-0.26.0.pom
          - https://illumon.jfrog.io/illumon/libs-local-jdk11/io/deephaven/authentication/0.26.0/authentication-0.26.0.pom
          - https://illumon.jfrog.io/illumon/libs-snapshot-local/io/deephaven/authentication/0.26.0/authentication-0.26.0.pom
        Required by:
            project : > project :Main > io.deephaven:deephaven-server:0.26.0 > io.deephaven:deephaven-psk-authentication-provider:0.26.0
...

Same thing for io.deephaven:Configuration:0.26.0 and io.deephaven:log-factory:0.26.0.

Note that they do exist locally w/ the "deephaven-" prefix, e.g. /Users/afarhangi/.m2/repository/io/deephaven/deephaven-authentication/0.26.0/deephaven-authentication-0.26.0.pom.

**Steps to reproduce**

**Expected results**

**Actual results**

**Additional details and attachments**

**Versions**
 - Deephaven: main, 0.26.0, 709f2a493f4e00d7ee7e3db11c5ed8e0e7b042f1
 - OS: Mac
 - Browser: N/A
 - Docker: Desktop 4.16.2, Engine 20.10.22, Compose v2.15.1

### PR DESCRIPTION
**Pull request by niloc132**
Does not address #4122, which would require changing shadow to use components correctly.

Fixes #4119

### CODE REVIEW

**Comment by niloc132** [comment]
This is not an ideal fix, but it does the job. I wasn't able to get the shadow plugin to be delayed in its changes of the default pom instance so that the GenerateMavenPom task is able to write it correctly, but instead just reimplements the logic to more closely match what we expect.

A "better" fix would delay our call to project.shadow.component() until after gradle has processed the artifactId changes to all dependency projects, propagating them into their own poms, so that they can be read. 

A "good" fix would be to provide a variant or a component just for the shadow. It appears that the shadow plugin attempts to do this, but doesn't really commit to it (giving us #4122).

Note that we also cannot update to shadow 8.1.1 at this time, as it requires gradle 8.

**Comment by devinrsmith** [comment]
Filed https://github.com/deephaven/deephaven-core/issues/4132 for follow-up.

File: buildSrc/src/main/groovy/io.deephaven.java-shadow-publishing-conventions.gradle [hunk 1]
```diff
@@ -19,8 +19,31 @@ tasks.withType(Javadoc) {
...
+      project.configurations.shadow.allDependencies.each {
+        if ((it instanceof ProjectDependency) || ! (it instanceof SelfResolvingDependency)) {
+          def dependencyNode = dependenciesNode.appendNode('dependency')
+          dependencyNode.appendNode('groupId', it.group)
+          BasePluginConvention base = it.dependencyProject.convention.getPlugin(BasePluginConvention)
+
+          dependencyNode.appendNode('artifactId', base.archivesBaseName)
+          dependencyNode.appendNode('version', it.version)
+          dependencyNode.appendNode('scope', 'runtime')
+          println dependencyNode
```
Comment (devinrsmith) [inline buildSrc/src/main/groovy/io.deephaven.java-shadow-publishing-conventions.gradle:42]: Remove.

**Review by devinrsmith — CHANGES_REQUESTED** [review]

**Comment by niloc132** [comment]
> Filed https://github.com/deephaven/deephaven-core/issues/4132 for follow-up.

Based on my read of the specific sources involved with this bug, this is unlikely to help.

**Review by devinrsmith — APPROVED** [review]

**Pull request merged**